src/mm-rating/mm-rating.js

```diff
@@ -0,0 +1,113 @@
+( function (scope) {
+    scope.Rating = Polymer({
+        is: 'mm-rating',
+        hovered: false,
+        properties: {
+            icon: {
+                type: String,
+                value: 'star'
+            },
+            customIcon: {
+                type: String,
+                value: null
+            },
+            iconWidth: {
+                type: Number,
+                value: 20
+            },
+            iconHeight: {
+                type: Number,
+                value: 20
+            },
+            rating: {
+                type: Number,
+                value: 0
+            },
+            displayTotals: {
+                type: Boolean,
+                value: false
+            },
+            displayGreetings: {
+                type: Boolean,
+                value: false
+            },
+            readonly: {
+                type: Boolean,
+                value: false
+            },
+            values: {
+                type: Object,
+                value: {0: 'Poor', 1: 'Need Improvement', 2: 'Reasonable', 3: 'Good', 4: 'Excellent'}
+            }
+        },
+
+        ready: function() {
+            this._defaultNoOfIcons = [];
+            if(typeof this.values === 'string'){
+                var greetObj = eval("("+ this.values +")");
```

Comment:
Here does'nt need extra property like 'range' because, rating is determine on the basis of number of objects. If you pass suppose 3 objects then the max star count will be 3.